### PR TITLE
Add makefile rules to push images to gcr staging repo.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind feature

**What this PR does / why we need it**:

Add makefile rules to push images to gcr staging repo.

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes-sigs/cloud-provider-azure/issues/233

**Special notes for your reviewer**:

**Release note**:
```
Add makefile rules to push images to gcr staging repo.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -158,3 +177,18 @@ endif
...
+
+$(PKG_CONFIG):
+	hack/pkg-config.sh > $@
+
+## --------------------------------------
+## Release
+## --------------------------------------
+
+.PHONY: deploy
+deploy: build-images hyperkube	push-images
```
Comment (feiskyer) [inline Makefile]: nit: remove extra spaces

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 1200s
+# this prevents errors if you don't use both _GIT_TAG and _PULL_BASE_REF,
+# or any new substitutions added in the future.
+options:
```
Comment (feiskyer) [inline cloudbuild.yaml]: any follow-ups to enable this build?

Comment (nilo19) [reply cloudbuild.yaml]: https://github.com/kubernetes/test-infra/pull/15545

**Pull request merged**